Add compose install engine (#14553)
docker/docker.github.io: compose/install.md

- On desktop systems like Docker Desktop for Mac and Windows, Docker Compose is
included as part of those desktop installs.

> - On Linux systems, you can install Docker Compose within the Docker Engine 

Reviewer: ```suggestion
- On Linux systems, you can install Docker Compose with the Docker Engine using the
```

> [convenience script](../engine/install/index.md#server){: target="_blank" rel="noopener" class="_"} for your distribution.  

Reviewer: ```suggestion
[convenience script](../engine/install/index.md#server){: target="_blank" rel="noopener" class="_"}. Select the install Docker Engine page for your distribution and then look for instructions on installing using the convenience script.
```

Otherwise, you should first install the [Docker Engine](../engine/install/index.md#server){: target="_blank" rel="noopener" class="_"}

> for your OS as described on the Get Docker page, then come back here for

Reviewer: ```suggestion
for your OS and then refer to this page for
```

## Install Compose

Follow the instructions below to install Compose on Mac, Windows, Windows Server
2016, or Linux systems.

> Install a different version
>

> > The instructions below outline installation of the current stable release

Reviewer 2: Wondering if we should replace / amend this to clearly mention these instructions are for v2, and link to github for v1 legacy install

    ```

    Then run the following command to download the current stable release of
    Compose ({{site.compose_version}}):

    ```powershell
    Invoke-WebRequest "https://github.com/docker/compose/releases/download/{{site.compose_version}}/docker-compose-Windows-x86_64.exe" -UseBasicParsing -OutFile $Env:ProgramFiles\Docker\docker-compose.exe
    ```

    >**Note**:  
    > On Windows Server 2019, you can add the Compose executable to `$Env:ProgramFiles\Docker`. Because this directory is  registered in the system `PATH`, you can run the `docker-compose --version` command on the subsequent step with no additional configuration.

    > To install a different version of Compose, substitute `{{site.compose_version}}` 
    > with the version of Compose you want to use.

3.  Test the installation.

    ```powershell

>     $ docker compose --version

Reviewer 3: Isn't it `docker compose version` instead ?

Reviewer 4: nice catch 👍

    Docker Compose version {{site.compose_version}}
    ```
</div>
<div id="linux" class="tab-pane fade" markdown="1">

### Install Compose on Linux systems

On Linux, you can download the Docker Compose binary from the

> [Compose repository release page on GitHub](https://github.com/docker/compose/releases){:target="_blank" rel="noopener" class="_"} and copying it into `$HOME/.docker/cli-plugins` as `docker-compose`.

Reviewer 4: maybe mention `/usr/local/lib/docker/cli-plugins` for global installation?

    ```console
    $ DOCKER_CONFIG=${DOCKER_CONFIG:-$HOME/.docker}
    $ mkdir -p $DOCKER_CONFIG/cli-plugins
    $ curl -SL https://github.com/docker/compose/releases/download/{{site.compose_version}}/docker-compose-linux-x86_64 -o $DOCKER_CONFIG/cli-plugins/docker-compose
    ```

    This command installs Compose for the active user under `$HOME` directory. To install Docker Compose for all users on your system, replace `~/.docker/cli-plugins` with `/usr/local/lib/docker/cli-plugins`.

    > To install a different version of Compose, substitute `{{site.compose_version}}`
    > with the version of Compose you want to use.

2. Apply executable permissions to the binary:

     ```console
    $ chmod +x $DOCKER_CONFIG/cli-plugins/docker-compose
    ```
    or if you choose to install Compose for all users
    ```console
    $ sudo chmod +x /usr/local/lib/docker/cli-plugins/docker-compose
    ```

3. Test the installation.

    ```console
    $ docker compose version
    Docker Compose version {{site.compose_version}}
    ```
</div>
<div id="linux-standalone" class="tab-pane fade" markdown="1">

### Install Compose as standalone binary on Linux systems

> You can use Compose as a standalone binary without the need of the Docker CLI.

Reviewer: ```suggestion
You can use Compose as a standalone binary without installing the Docker CLI.
```

1. Run this command to download the current stable release of Docker Compose:

  ```console
  $ curl -SL https://github.com/docker/compose/releases/download/{{site.compose_version}}/docker-compose-linux-x86_64 -o /usr/local/bin/docker-compose
  ```

  > To install a different version of Compose, substitute `{{site.compose_version}}`
  > with the version of Compose you want to use.

2. Apply executable permissions to the binary:

  ```console
  $ sudo chmod +x /usr/local/bin/docker-compose
  ```
  
  > **Note**:
  >
  > If the command `docker-compose` fails after installation, check your path.
  > You can also create a symbolic link to `/usr/bin` or any other directory in your path.
  >
  > For example:
  > ```console
  > $ sudo ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose
  > ```

3. Test the installation.

    ```console
    $ docker-compose --version
    Docker Compose version {{site.compose_version}}
    ```
</div>
</div>

----

## Uninstallation

To uninstall Docker Compose if you installed using `curl`:

  ```console
    $ rm $DOCKER_CONFIG/cli-plugins/docker-compose

>   ```

Reviewer 2: Indentation on these looks "off"; when using code-fences, they should be without indentation (and no indentation within the block itself)

    
  or if you choose to install Compose for all users

>   ```console

Reviewer: nit: add a blank line before the code block

    ```

    Then run the following command to download the current stable release of
    Compose ({{site.compose_version}}):

    ```powershell
    Invoke-WebRequest "https://github.com/docker/compose/releases/download/{{site.compose_version}}/docker-compose-Windows-x86_64.exe" -UseBasicParsing -OutFile $Env:ProgramFiles\Docker\docker-compose.exe
    ```

>     >**Note**:  
>     > On Windows Server 2019, you can add the Compose executable to `$Env:ProgramFiles\Docker`. Because this directory is  registered in the system `PATH`, you can run the `docker-compose --version` command on the subsequent step with no additional configuration.

Reviewer 2: Looks like this is using trailing whitespace to force a line break; better to add an explicit newline to prevent editors from accidentally removing them;

```suggestion
    > **Note**
    >
    > On Windows Server 2019, you can add the Compose executable to `$Env:ProgramFiles\Docker`. Because this directory is  registered in the system `PATH`, you can run the `docker-compose --version` command on the subsequent step with no additional configuration.
```

>     > To install a different version of Compose, substitute `{{site.compose_version}}` 

Reviewer 2: I think the trailing whitespace may cause a line break to be forced; don't think that was intentional, so better to remove it (also see above)

```suggestion
    > To install a different version of Compose, substitute `{{site.compose_version}}`
```

You can use Compose as a standalone binary without installing the Docker CLI.

1. Run this command to download the current stable release of Docker Compose:

  ```console
  $ curl -SL https://github.com/docker/compose/releases/download/{{site.compose_version}}/docker-compose-linux-x86_64 -o /usr/local/bin/docker-compose
  ```

  > To install a different version of Compose, substitute `{{site.compose_version}}`
  > with the version of Compose you want to use.

2. Apply executable permissions to the binary:

  ```console
  $ sudo chmod +x /usr/local/bin/docker-compose
  ```
  
  > **Note**:
  >
  > If the command `docker-compose` fails after installation, check your path.
  > You can also create a symbolic link to `/usr/bin` or any other directory in your path.
  >
  > For example:
  > ```console
  > $ sudo ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose
  > ```

3. Test the installation.

    ```console
    $ docker-compose --version
    Docker Compose version {{site.compose_version}}
    ```
</div>
</div>

----

## Uninstallation

To uninstall Docker Compose if you installed using `curl`:

  ```console
  $ rm $DOCKER_CONFIG/cli-plugins/docker-compose
  ```
    

>   or if you choose to install Compose for all users  

Reviewer 2: Canyon remove the leading whitespace on these lines?
